### ISSUE TITLE
GSO seems to break wireguard dialing

### ISSUE THREAD

**Issue by Mahdi-zarei** [SagerNet/sing-box#1222]
### Operating system

Linux

### System version

Ubuntu 22.04.3 LTS (GNU/Linux 5.15.0-86-generic x86_64)

### Installation type

Original sing-box Command Line

### Version

```shell
latest dev-next up until 1.8.0-alpha.17
```

### Description

When the GSO for wireguard is turned on, dials fail and practically nothing works, with commenting the gso: true line everything works fine. The tun gso seems to work, even though there are some error logs.

### Reproduction

server config: 
```
{
  "log": {
    "level": "error",
    "timestamp": true
  },
  "dns": {
    "disable_cache": false,
    "servers": [
      {
        "tag": "google",
        "address": "8.8.8.8",
        "strategy": "prefer_ipv4",
        "detour": "direct"
      }
    ]
  },
  "inbounds": [
    {
      "type": "tun",
      "tag": "TtoWG",
      "interface_name": "sing",
      "inet4_address": "172.19.0.1/24",
      "inet6_address": "fdfe:dcba:9876::1/96",
      "mtu": 1500,
      "gso": true,
      "sniff": true,
      "auto_route": true,
      "stack": "system",
      "include_interface": [
        "tun0"
      ]
    },
    {
      "type": "vless",
      "tag": "Vin6",
      "listen": "...",
      "listen_port": 443,
      "domain_strategy": "prefer_ipv4",
      "sniff": true,
      "users": [
        {
          ...
        }
      ],
      "multiplex": {
        "enabled": true
      },
      "tls": {
        ...
      }
    },
    {
      "type": "vless",
      "tag": "Vin4",
      "listen": "...",
      "listen_port": 443,
      "domain_strategy": "prefer_ipv4",
      "sniff": true,
      "users": [
        ...
      ],
      "multiplex": {
        "enabled": true
      },
      "tls": {
        ...
      }
    }
  ],
  "outbounds": [
    {
      "type": "direct",
      "tag": "direct"
    },
    {
      "type": "block",
      "tag": "block"
    },
    {
      "type": "dns",
      "tag": "dns-out"
    },
    {
      "type": "wireguard",
      "tag": "wg",
      "server": "...",
      "server_port": 2408,
      "system_interface": true,
#      "gso": true,
      "interface_name": "wg0",
      "local_address": [
        "172.16.0.2/24",
        "2606:4700:110:8ea5:e3d2:d16c:3d44:b254/96"
      ],
      "mtu": 1280,
      "peer_public_key": "...",
      "private_key": "...",
      "reserved": "..."
    }
  ],
  "route": {
    "rule_set": [
      {
        ...
      }
    ],
    "rules": [
      {
        "ip_is_private": true,
        "outbound": "block"
      },
      {
        "protocol": "dns",
        "outbound": "dns-out"
      }
    ]
  }
}
```

uncomment the gso, and the log fills with dial tcp xxx: i/o timeout and connectivity will be lost.

### Logs

```shell
inbound/vless[Vin4]: process connection from 91.98.43.90:12805: dial tcp 149.154.175.100:80: i/o timeout
inbound/vless[Vin6]: process connection from [2a01:5ec0:5814:2d75:13ce:e8b2:11fc:c8b2]:33516: dial tcp [2001:b28:f23d:f003::a]:443: i/o timeout
inbound/tun[TtoWG]: read packet: too many segments
```

**Comment by nekohasekai** [comment]
Try 1.8.0-beta.1

**Comment by Mahdi-zarei** [comment]
updated to the latest commit, the issue still persists, cannot dial with wireguard outbound when GSO is active.

**Comment by Mahdi-zarei** [comment]
Currently this issue is fixed as of d69a7ca4fdc1e4caf5c0f0d283c901bcf7639b22 and with GSO active, TUN and wireguard work properly.
